otherlibs/stdune/src/platform.ml

```diff
@@ -1,29 +1,68 @@
 module OS = struct
   (* CR-someday alizter: Include mingw32, mongw64, cygwin *)
   type t =
-    | Darwin
-    | Linux
-    | Windows
-    | FreeBSD
-    | NetBSD
-    | OpenBSD
-    | Other
+    [ `Darwin
+    | `Linux
+    | `Windows
+    | `FreeBSD
+    | `NetBSD
+    | `OpenBSD
+    | `DragonFly
+    | `Haiku
+    | `Serenity
+    | `Solaris
+    | `Other
+    ]
+
+  (* Include all constructors here. *)
+  let all =
+    [ `Darwin
+    ; `Linux
+    ; `Windows
+    ; `FreeBSD
+    ; `NetBSD
+    ; `OpenBSD
+    ; `DragonFly
+    ; `Haiku
+    ; `Serenity
+    ; `Solaris
+    ; `Other
+    ]
+  ;;
+
+  type fswatch_support =
```

Comment:
This type is unnecessary because it's the equivalent of just writing a pattern match.

Also, it should never be defined here.